### PR TITLE
fix!: Rename domains `c_name` to `cname`

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Related to https://github.com/cloudquery/cloudquery/issues/2189.

1. Added codegen drift detection
2. Moved generated resources under `../resources/services` for consistency with other plugins (there's a follow up here to put each namespace in a separate directory, but didn't want to do too many changes)
3. Moved `plugin/plugin.go` to `resources/plugin/plugin.go`. This is required so Go Releaser embeds the correct version of the plugin during release, see https://github.com/cloudquery/cloudquery/blob/6ab042925afaf7fe0a7fccbedd17f0915d40f7cf/plugins/.goreleaser.yaml#L12

### CODE REVIEW

File: .github/workflows/source_heroku.yml [hunk 1]
```diff
@@ -40,3 +40,5 @@ jobs:
         run: make gen-docs
       - name: Fail if docs are changed
         run: test "$(git status -s ./docs/tables | wc -l)" -eq 0
```
Comment (hermanschaaf) [inline .github/workflows/source_heroku.yml:42]: I think this is missing the `gen-code` step

**Review by hermanschaaf — CHANGES_REQUESTED** [review]
Missing gen-code, otherwise LGTM 👍

File: .github/workflows/source_heroku.yml (hunk 1 above)
Comment (erezrokah) [reply .github/workflows/source_heroku.yml:42]: Ah, good catch, fixed in https://github.com/cloudquery/cloudquery/pull/2635/commits/4ec26f7e298e5c870e1d68bb66853b4bc178067b

**Review by hermanschaaf — APPROVED** [review]
Nice, thanks for doing the refactoring work as well 👍

**Comment by erezrokah** [comment]
@hermanschaaf looks like another breaking change due to https://github.com/cloudquery/cloudquery/pull/2635/files#diff-f8e4debb63fb716d5f8829d87f8dbdc19564da7bc53b4d3481d6cade51593fab:
image

**Pull request merged**